src/services/orderbook_service.ts

```diff
@@ -70,6 +71,136 @@ export class OrderBookService {
         };
     }
 
+    public checkBidsOrAsks = (order: SRAOrder, req: OrderbookPriceRequest): boolean => {
+        if (req.taker !== NULL_ADDRESS && req.taker !== order.order.taker) {
+            return false;
+        }
+        if (req.feeRecipient !== NULL_ADDRESS && order.order.feeRecipient.toLowerCase() !== req.feeRecipient) {
+            return false;
+        }
+        if (req.takerTokenFee !== 0) {
+            if (Number(order.order.takerTokenFeeAmount.toString()) !== Number(order.order.takerAmount) * req.takerTokenFee / 100000) {
```

Comment:
This is not using BigNumber math and will result in rounding errors. If the order creator used BigNumber math (which they should), then you won't have a match with that math logic

Reply:
We can't use `parseUnits` in this project because the `ethers` version is lower than the frontend part. I solved this problem.